src/main/java/net/neoforged/neoforge/client/renderstate/RenderStateExtensions.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.renderstate;
+
+import com.google.common.collect.ImmutableList;
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import it.unimi.dsi.fastutil.objects.Reference2ObjectArrayMap;
+import it.unimi.dsi.fastutil.objects.Reference2ObjectOpenHashMap;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import net.minecraft.client.renderer.entity.EntityRenderer;
+import net.minecraft.client.renderer.entity.state.EntityRenderState;
+import net.minecraft.client.renderer.state.MapRenderState;
+import net.minecraft.core.Holder;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.saveddata.maps.MapDecorationType;
+import net.minecraft.world.level.saveddata.maps.MapItemSavedData;
+import org.jetbrains.annotations.ApiStatus;
+
+public final class RenderStateExtensions {
+    private RenderStateExtensions() {}
+
+    private static final Map<Class<?>, Collection<BiConsumer<?, ?>>> ENTITY = new Reference2ObjectArrayMap<>();
+    private static final Map<Class<?>, Collection<BiConsumer<?, ?>>> ENTITY_CACHE = new Reference2ObjectOpenHashMap<>();
```

Comment:
Is there a reason this uses `Reference2ObjectArrayMap`? There's a fair number of entity renderer subclasses in vanilla alone (22) -- and this `ENTITY_CACHE` will have an entry for each of them, meaning that the worst-case you are iterating the entire map, and on average you're iterating 11 things, plus whatever mods add. It feels strange that an array map would be faster at this than an identity hash map? Please correct me if I'm wrong.

Reply:
The `ENTITY` map is only ever iterated outside of registration and the `ENTITY_CACHE` is a `Reference2ObjectOpenHashMap` and is the one that is actually queried when a renderer computes the render state.

Reply:
Nevermind, I misread this and was looking at the line above